### PR TITLE
Bugfix: virthost would not deploy correctly

### PR DESCRIPTION
**Pull request by moio**
## What does this PR change?

Prevent deploy errors of `min-kvm` as per the default testsuite example file for libvirt.

Specifically, the libvirt daemon would not start:

```
----------
          ID: libvirtd_service
    Function: service.running
        Name: libvirtd
      Result: False
     Comment: The named service libvirtd is not available
```

Moreover, `default-net_destroyed` would fail if the network is there but not active.

### CODE REVIEW

**Comment by moio** [comment]
@cbosdo, @Bischoff: why are those states completely broken, don't we use them in CI tests? Do we still care about having them in sumaform?

Additionally: don't we have salt modules to deal with pools/nets/etc?

https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.virt.html

Why do we use `virsh` in `cmd.run` here?

**Comment by cbosdo** [comment]
> @cbosdo, @Bischoff: why are those states completely broken, don't we use them in CI tests? Do we still care about having them in sumaform?

We still use those in the testsuite... it's pretty weird that those are failing for you: I'm still using it at every new deployment of my test / dev systems here. Those seem like corner cases that prevent the state from being repeatable, but should  work with a system built from scratch like what we do in the CI.

> Additionally: don't we have salt modules to deal with pools/nets/etc?
> 
> https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.virt.html
> 
> Why do we use `virsh` in `cmd.run` here?

Because the virt modules for that in salt are still broken. I submitted fixes upstream and to our packages, but those are not yet in.

File: salt/virthost/init.sls [hunk 1]
```diff
@@ -7,6 +7,7 @@ virthost_packages:
         {% if '15' in grains['osrelease'] %}
         - patterns-server-kvm_server
         - python3-six  # Workaround missing virt-manager-common dependency
+        - libvirt
```
Comment (cbosdo) [inline salt/virthost/init.sls]: Install the whole `libvirt` with all driver is may be not the best thing to do, better install the `libvirt-daemon-qemu` package.

What distro doesn't pull the corresponding libvirt drivers when installing the `patterns-server-kvm_server`? Is it in the `kvm_tools` pattern instead?

File: salt/virthost/init.sls [hunk 2]
```diff
@@ -83,7 +84,7 @@ default-net.xml:
 default-net_destroyed:
   cmd.run:
     - name: 'virsh net-destroy default'
-    - onlyif: 'virsh net-dumpxml default'
+    - onlyif: 'virsh net-info default | grep Active | grep yes'
```
Comment (cbosdo) [inline salt/virthost/init.sls]: ACK

File: salt/virthost/init.sls (hunk 1 above)
Comment (cbosdo) [reply salt/virthost/init.sls]: Just checked here on an openSUSE: the `kvm_server` pattern includes the `libvirt-daemon-qemu` package, which should drag the needed libvirt dependencies

**Comment by Bischoff** [comment]
Cédric answered.

File: salt/virthost/init.sls (hunk 1 above)
Comment (moio) [reply salt/virthost/init.sls]: This was tested on 15 SP1 which is the default in the testsuite example:

https://github.com/uyuni-project/sumaform/blob/master/main.tf.libvirt-testsuite.example#L134

This does not drag in enough dependencies, as the `libvirtd` service is not there.

Should I report this as a libvirt bug?

Meanwhile I changed to `libvirt-daemon-qemu` and can confirm that one is sufficient at least on SLE 15 SP1.

Comment (cbosdo) [reply salt/virthost/init.sls]: sounds like a SLE bug indeed.
